# Meta
allow_none: True
arduino_use_yaml: true

# Global
source: "my_rosbag" # must be "arduino", "my_rosbag", or "nothing"

# Experimental setup description
length_arm: 0.650

imus: 
  positions:
    - 0.0000 # Don't remove this, it corresponds to the base's orientation
    - 0.0622
    - 0.1422
    - 0.2222
    - 0.3022
    - 0.3822
    - 0.4622
    - 0.5422
    - 0.6222
  offset_euler_angles:
    roll: 1.57079632679
    pitch: 0
    yaw: 0

segments: 
- 4
- 4
- 2
- 1
- 5
- 2
- 4
- 2

segments_radius: 0.025

markers: 
  use: True
  positions: #None
  - 0.360
  - 0.600
  show: True
  radius: 0.04 # For RVIZ

base_orientation_quaternion: 
  x: 0
  y: 0
  z: 1
  w: 0


# Arduino
arduino: 
  boot: false
  port: "/dev/ttyACM0"
  baud: 115200
  sampling_rate_ms: 10

# My rosbag 
my_rosbag:
  to_save: 
    quaternions: false
    markers: false
    accelerometers: false
    file: "freshly_recorded_data.bag"
  to_play:
    rate: 1
    start: 0
    file: "prev_saved_data_to_play.bag"


# Calibration
calib: 
  ## Use saved calibration
  use_saved: false
  saved_path: "~/catkin_ws/src/sesa_main_folder/sesa/saved/calibs/"
  saved_file: "my_calib.yaml"
  ## Measure new calibration
  calibrate: false
  calibrate_path: "~/catkin_ws/src/sesa_main_folder/sesa/saved/calibs/"
  calibrate_file: "my_calib.yaml"



# Visualization
rviz_gui: true



